profiling/src/profiling/mod.rs

```diff
@@ -329,6 +412,150 @@ impl TimeCollector {
             }
         }
 
+        #[cfg(all(target_os = "linux", feature = "io_profiling"))]
+        if let (Some(socket_read_time_offset), Some(socket_read_time_samples_offset)) =
+            (socket_read_time_offset, socket_read_time_samples_offset)
+        {
+            let upscaling_info = UpscalingInfo::Poisson {
+                sum_value_offset: socket_read_time_offset,
+                count_value_offset: socket_read_time_samples_offset,
+                sampling_distance: SOCKET_READ_TIME_PROFILING_INTERVAL.load(Ordering::SeqCst),
+            };
+            let values_offset = [socket_read_time_offset, socket_read_time_samples_offset];
+            match profile.add_upscaling_rule(&values_offset, "", "", upscaling_info) {
+                Ok(_id) => {}
+                Err(err) => {
+                    warn!("Failed to add upscaling rule for socket read time samples, socket read time samples reported will be wrong: {err}")
+                }
+            }
+        }
```

Comment:
this could be put into a macro / function call to save on LOC and make it more maintainable